### PR TITLE
Post install script

### LINKED ISSUE #67: post install script
I was compiling native libraries and everything needed to make camanJS work with AWS lambda (caman, canvas, fibers) and I needed a way to copy compiled modules after `npm install --production`, so I created a [fork](https://github.com/navihtot/node-lambda) of this project which allows you to have  `post_install.sh ` which (if exists) is runned after `npm install --production` and before creating a .zip and deploying to lambda.
There is also the npm package  `node-lambda-postinstall` if it helps anyone.
It would be nice to have this option in original project :)

### PR DESCRIPTION
**Pull request by navihtot**
If file `post_install.sh` exists than it is run after `npm install --production` and before creating .zip and deploying to AWS Lambda

Fixes #67

### CODE REVIEW

**Comment by DeviaVir** [comment]
Hi @navihtot!

Thanks for your contribution! Could you explain the usecase and add a bit of documentation on how others might use this function?

**Comment by navihtot** [comment]
Hi, my usecase is that I needed to use CamanJS library on Lambda (which is dependent on node-canvas and fibers). The problem is that canvas needs native libraries compiled on same version of Amazon Linux that lambda runs on, and modules needed to be compiled with node-gyp after that.
So I needed a way that after npm install I can pull my [precompiled libs and modules](https://github.com/navihtot/node-caman-aws-lambda) and overwrite existing ones in node_modules folder.

But I presume someone could have a need to do any other action with .sh script after npm install and before deployment.
Usage is simple, if script with that name exists its runs else its skipped. 

I'll try to add some simple documentation later today.

Btw - lambda now finally supports newer node version 4.3.2., and I tried to deploy with `AWS_RUNTIME=nodejs4.3` in `.env` - it works!

**Comment by navihtot** [comment]
There is also a npm -> https://www.npmjs.com/package/node-lambda-postinstall

**Comment by navihtot** [comment]
Here, added section **Post install script** with short description in readme.md

File: README.md [hunk 1]
```diff
@@ -102,6 +102,11 @@ $ node-lambda deploy --help
 
 AWS Lambda doesn't let you set environment variables for your function, but in many cases you will need to configure your function with secure values that you don't want to check into version control, for example a DB connection string or encryption key. Use the sample `deploy.env` file in combination with the `--configFile` flag to set values which will be prepended to your compiled Lambda function as `process.env` environment variables before it gets uploaded to S3. 
 
+## Post install script
+
+When running `node-lambda deploy` if you need to do some action after `npm install --production` and before deploying to AWS Lambda (i.e. replace some modules with precompiled ones or download some libraries) you can create `post_install.sh` script. If the file exists the script will be runned (and output shown after execution) if not it is skipped. Make sure that the script is executable.
```
Comment (DeviaVir) [inline README.md]: `the script will be runned` => `the script will be executed`

**Comment by DeviaVir** [comment]
Could you rebase this branch against current master so it's good to merge? I'm onboard with the changes :+1:

**Comment by navihtot** [comment]
not sure how to do this. i get many conflicts... i could create another fork/branch from current master and implement it there. but wont have time for that until the end of the week.

**Comment by DeviaVir** [comment]
@navihtot after you've merged in https://github.com/navihtot/node-lambda/pull/1 into your branch, you can rebase by doing the following on your machine:

```
git remote add upstream git@github.com:motdotla/node-lambda.git
git fetch upstream
git checkout post-install-script
git rebase -i upstream/master
```
from there you'll see a `vi` editor where you can pick specific commits, replace all "pick"'s  by "squash" except the first one. After that is completed and you have set the commit message, you can run `git push --force` and your commits will be squashed ending up in a much nicer commit history in this repo.

**Comment by navihtot** [comment]
tried it. i still get too much conflicts when i do rebase :( should I just create fork again an add functionality from scratch ?

**Comment by DeviaVir** [comment]
@navihtot that's probably the fastest way if you don't want to deal with the merge conflicts now. If you set that up and create the PR I'll get it merged ASAP so you don't have to deal with any conflicts later on.

**Comment by navihtot** [comment]
here I created another pull request #78

**Pull request closed without merging**